src/main/scala/org/apache/spark/streamdm/tasks/ClusteringTrainEvaluate.scala

```diff
@@ -55,7 +55,23 @@ class ClusteringTrainEvaluate extends Task {
    * @param ssc The Spark Streaming context in which the task is run.
    */
   def run(ssc:StreamingContext): Unit = {
+    //check if clusterer is streamkm
+    if(this.clustererOption.getValue().toString.contains("StreamKM")){
```

Comment:
After you move the code from `trainStreamKM(input: DStream[Example]):  DStream[(Example, Double)]` to `assign(input: DStream[Example]): DStream[(Example,Double)]` then it won't be necessary to have this change to `ClusteringTrainEvaluate.scala`.